client/session.go

```diff
@@ -683,28 +685,19 @@ func (s *session) Write(namespace, id string, t time.Time, value float64, unit x
 
 	for i := range state.queues {
 		if err := state.queues[i].Enqueue(state.op); err != nil {
-			state.Unlock()
-			state.decRef()
-
-			// NB(r): if this ever happens we have a code bug, once we
-			// are in the read lock the current queues we are using should
-			// never be closed
-			s.RUnlock()
-			s.opts.InstrumentOptions().Logger().Errorf("failed to enqueue write: %v", err)
-			return err
+			// NB(r): if this happens we have a bug, once we are in the read
+			// lock the current queues should never be closed
+			panic(fmt.Sprintf("failed to enqueue write: %v", err))
 		}
 
 		state.incRef()
 		enqueued++
 	}
 
-	// Wait for writes to complete. We don't need to loop over Wait() since there
-	// are no spurious wakeups in Golang and the condition is one-way and binary.
```

Comment:
It seems like it's just explaining how `.Wait()` works, am I missing something subtle? I'd expect a comment if we were looping over some variable, not if we do the default Go thing.